src/lib/data/html/speeddial.html

```diff
@@ -903,19 +988,68 @@
 }
 </script>
 </head>
-
 <body>
 
 <noscript>%JAVASCRIPT-DISABLED%</noscript>
 
-<div id="quickdial">
+<div class="SearchBar" id="SearchBar">
+    <div class="SearchBar" id="SearchBarToggle" onClick="searchToggle()">
+    </div>
+        <div id="SearchForm">
+            <input id="SearchInput" name="" value="%SEARCH-HINT%" onfocus="searchFocus()" onfocusout="searchRelease()">
+            <input id="submit" type="button" value="" onClick="Search()">
+        </div>
+     <div style="width: 36px"></div>
 </div>
 
-<a onClick="configureSpeedDial()" title="%SETTINGS-TITLE%" class="sett"></a>
+<a onClick="configureSpeedDial()" title="%SETTINGS-TITLE%" class="sett" id="sett"></a>
 <a onClick="addSpeedDial()" title="%ADD-TITLE%" class="add"></a>
+<div id="quickdial">
+</div>
+
 
 <script type="text/javascript">
 
+if ("%IMG_BACKGROUND%" == '') {
+    document.getElementById("html").style.background = 'url("qrc:html/sd_bg.svg")';
+    document.getElementById("html").style.backgroundSize = 'cover';
+}
+
+if (!%SEARCH-ENABLED%) {
+    document.getElementById('quickdial').style.margin = '10px auto';
+    document.getElementById('SearchBar').style.width = '40px';
+    document.getElementById('SearchBar').style.background = 'rgba(0,0,0, 0.0)';
+    document.getElementById('sett').style.top = '4px';
+    document.getElementById('SearchBarToggle').style.background = 'url("qrc:html/search.svg") no-repeat center center, rgba(65,65,65, 0.0)';
+    document.getElementById('SearchForm').style.visibility = 'hidden';
+}
+
+function searchFocus() {
+    document.getElementById('SearchInput').style.color = 'rgba(236,236,236, 1.0)';
+    document.getElementById('SearchInput').value = '';
+}
+
+function searchRelease() {
+    document.getElementById('SearchInput').style.color = 'rgba(118,118,118, 1.0)';
+    document.getElementById('SearchInput').value = "%SEARCH-HINT%";
+}
+
+function Search() {
+    if (document.getElementById('SearchInput').value == "%SEARCH-HINT%") {
+        value = ''
+    } else {
+        value = document.getElementById('SearchInput').value
+    }
+    var url = '%SEARCH-URL%'.replace('%s', value);
```

Comment:
You can't expose search url to JavaScript as there is also support for `POST` search engines which will not work this way.
You should just expose `search(string)` C++ function to SpeedDial, which will handle the searching.

Reply:
> POST search engines 

I completely forgot about them %)
> which will not work this way.

Why not? It can be simply converted to GET request...
```
.........
    page.replace(QLatin1String("%SEARCH-POST%"), dial->searchEngine().postData);
.........
```
```
.........
var SEARCH_POST = "%SEARCH-POST%";
.........
    if (SEARCH_POST) {
        SEARCH_URL = SEARCH_URL + '?' + SEARCH_POST; 
    }
.........
```
Or as a variant we can generate  different requests depended on engine method.

Reply:
No, this is not how you make POST request from JavaScript. One way to do it is to create temporary `<form>` and accept it.

In any case, that logic is already implemented and tested on C++ side so there is no reason to duplicate it in speed dial.